src/MultiOwnable.sol

```diff
@@ -1,118 +1,206 @@
+    ///
+    /// @dev Intended to be called when the smart account is first deployed.
+    /// @dev Reverts if a provided owner is neither 64 bytes long (for passkey) nor a valid address.
+    ///
+    /// @param owners The intiial list of owners to register.
     function _initializeOwners(bytes[] memory owners) internal virtual {
         for (uint256 i = 0; i < owners.length; i++) {
             if (owners[i].length != 32 && owners[i].length != 64) {
                 revert InvalidOwnerBytesLength(owners[i]);
             }
+
             if (owners[i].length == 32 && uint256(bytes32(owners[i])) > type(uint160).max) {
                 revert InvalidEthereumAddressOwner(owners[i]);
             }
-            _addOwnerAtIndex(owners[i], _getMultiOwnableStorage().nextOwnerIndex++);
+
+            _addOwnerAtIndex({owner: owners[i], index: _$().nextOwnerIndex++});
         }
     }
 
+    /// @notice Convenience function used to add the first 255 owners.
+    ///
+    /// @param owner The owner raw bytes to add.
     function _addOwner(bytes memory owner) internal virtual {
-        _addOwnerAtIndex(owner, _getMultiOwnableStorage().nextOwnerIndex++);
+        _addOwnerAtIndex({owner: owner, index: _$().nextOwnerIndex++});
     }
 
+    /// @notice Adds an owner at the given `index`.
+    ///
+    /// @dev Reverts if `owner` is already registered as an owner.
+    ///
+    /// @param owner The owner raw bytes to register.
+    /// @param index The index to write to.
     function _addOwnerAtIndex(bytes memory owner, uint256 index) internal virtual {
         if (isOwnerBytes(owner)) revert AlreadyOwner(owner);
 
-        _getMultiOwnableStorage().isOwner[owner] = true;
-        _getMultiOwnableStorage().ownerAtIndex[index] = owner;
+        _$().isOwner[owner] = true;
+        _$().ownerAtIndex[index] = owner;
 
-        emit AddOwner(index, owner);
+        emit AddOwner({index: index, owner: owner});
     }
 
-    /// @dev There is no logic in this contract
-    /// to allow for address(this) to be msg.sender.
-    /// This should be enabled in the inheriting contract
-    /// to allow for a passkey owner to call these functions.
+    /// @notice Checks if the sender is an owner of this contract or the contract itself.
+    ///
+    /// @dev Revert if the sender is not an owner fo the contract itself.
     function _checkOwner() internal view virtual {
-        if (!isOwnerAddress(msg.sender)) if (msg.sender != address(this)) revert Unauthorized();
+        if (isOwnerAddress(msg.sender) || (msg.sender == address(this))) {
+            return;
+        }
+
+        revert Unauthorized();
     }
 
-    function _getMultiOwnableStorage() internal pure returns (MultiOwnableStorage storage $) {
+    /// @notice Helper function to get a storage reference to the `MultiOwnableStorage` struct.
+    ///
+    /// @return $ A storage reference to the `MultiOwnableStorage` struct.
+    function _$() internal pure returns (MultiOwnableStorage storage $) {
```

Comment:
is this idiomatic?

Reply:
Not sure. I personally use `_s()` and `s` but I saw you were already using `_$()` as naming for the slot ( and I find `_getMultiOwnableStorage()` a bit long to write each time)

Reply:
Though definitely shorter, if we're claiming to follow ERC-7201 I think we should follow all of its conventions.

Reply:
Done in #bc2496b.